comment/tests/test_models.py

```diff
@@ -110,6 +112,24 @@ def test_has_resolved_state(self, mocked_hasattr):
         mocked_hasattr.return_value = False
         self.assertFalse(comment.has_resolved_state)
 
+    def test_urlhash_is_unique(self):
+        comment_1 = self.child_comment_1
+        comment_2 = self.child_comment_2
+        comment_1.urlhash = None
+
+        with patch(
+            'comment.managers.CommentManager.generate_urlhash',
+            side_effect=[True, False]
+        ):
+            comment_1.save()
+
+            self.assertEqual(comment_1.urlhash, True)
+
+        comment_1.urlhash = None
+        comment_1.save()
+        comment_1.refresh_from_db()
+        self.assertNotEqual(comment_1.urlhash, comment_2.urlhash)
```

Comment:
This is just a workaround to push the changes. I tried different sort of stuff to fix this but was unable to do it till now. I have give this close to 3 hours and am certainly out of patience now. It would be nice if you could help and fix this.

Reply:
The test helped us finding an issue with the current while statement, I already requested change with suggestion.
Here you go:

```python
    @patch('comment.managers.CommentManager.generate_urlhash')
    def test_urlhash_is_unique(self, mocked_generate_urlhash):
        mocked_generate_urlhash.side_effect = ['first_urlhash', 'first_urlhash', 'second_urlhash']
        first_comment = self.create_comment(self.content_object_1)
        self.assertEqual(first_comment.urlhash, 'first_urlhash')
        mocked_generate_urlhash.assert_called_once()
        second_comment = self.create_comment(self.content_object_1)
        self.assertEqual(second_comment.urlhash, 'second_urlhash')
```

Reply:
I just tested your code. It is not passing on my machine. Its giving `StopIteration` which is probably when it is not able to set the given value.

Reply:
yes bcz you didn't change while loop issue, as I suggested

Reply:
Okay it passed after using your suggestion. I'm really not sure why.

Reply:
because `self.__class__.objects.filter(urlhash=hash_val).exists()` is always True, you don't update `hash_val` inside while loop.

Reply:
got it, this was made before you explained in the conversation thread on the suggestion on the `model.`